lib/system/supports-colors.js

```diff
@@ -31,9 +31,8 @@ module.exports = (function () {
     return false;
   }
 
-  if (argv.indexOf('--color') !== -1 ||
-    argv.indexOf('--color=true') !== -1 ||
-    argv.indexOf('--color=always') !== -1) {
+  if (!argv.indexOf('--no-color') !== -1 ||
+    !argv.indexOf('--color=false') !== -1) {
```

Comment:
This is not how you invert an logical condition. I already described it in detail above.